### DISCUSSION TITLE
Storage resources are insufficient for the write operation after adding additional disk pool.

### DISCUSSION THREAD

**Discussion by AlfLjung** [minio/minio#17875] [Q&A] [answered]
Hi,
After expanding a distributed cluster with an additional disk pool (including 4 new servers) we see following errors in the message-log file.

Error seen using command : sudo journalctl -r -u t7prd-minio.service
API: SYSTEM()
       1: cmd/erasure.go:427:cmd.erasureObjects.nsScanner.func1()
       2: internal/logger/logger.go:258:logger.LogIf()
Error: Storage resources are insufficient for the write operation .minio.sys/buckets/.usage-cache.bin (cmd.InsufficientWriteQuorum)

Restarting the minio cluster or the systemd service on the particular minio server resolve the issue on the server that got the errors, but the error message seems to move to another node shortly after. Is this  error something we shall ignore and is common when adding a disk pools and will disappear ?

We use minio release : 2023-04-07T05:28:58Z
The error is not visible in the minio console log or when typing command 'mc admin logs <ALIAS>'

Disk pool info
Pools:
   1st, Erasure sets: 12, Drives per erasure set: 16   <---- initial pool
   2nd, Erasure sets: 6, Drives per erasure set: 16   <----- new pool

Regards

**Comment by klauspost** [comment, marked as answer]
Please upgrade. This has been fixed in later versions. The issue is stale disk information, so a reboot usually temporarily fixes it.

**Reply by AlfLjung** [reply]
Hi,
Thanks for the information.
Regards
/Alf

**Comment by mshanmu** [comment]
@klauspost , if you don't mind, can you please point me to the bug or PR for this issue?

**Reply by klauspost** [reply]
https://github.com/minio/minio/pull/17085

**Reply by mshanmu** [reply]
Thanks, @klauspost . I suspect the issue I am seeing matches with this PR, but, since, the PR does not have any associated bug/scenarios, let me explain what I am seeing. 

1. Out of 16 MinIO pods, one of the pods suddenly start throwing "**503 SlowDown Resource requested is unreadable, please reduce your request rate**" for all Multipart Put requests. Of course, this is because the disk info is throwing a write quorum error as seen by the log message, which was not really helpful.
```
"error": {
   "message": "*errors.errorString",
   "source": [
      "internal/logger/logger.go:258:logger.LogIf()",
      "cmd/erasure-server-pool-go:1418:cmd.(*erasureServerPools).NewMultipartUpload()",
     ....
   ]
}
```
2. Once, it is stuck in this state, it never comes out of it, unless you do a minio restart or pod deletion.
3. Of course, all other pods work fine. 

Does the above scenario is resolved with this PR?

**Reply by klauspost** [reply]
Sounds similar. Either way, just upgrade.

**Reply by Vinaum8** [reply]
@beck-8 @klauspost i am using quay.io/minio/minio:RELEASE.2024-07-16T23-46-41Z and my error logs continue: 

`component=ingester org_id=fake msg="failed to flush" err="failed to flush chunks: store put chunk: SlowDownWrite: Resource requested is unwritable, please reduce your request rate`

any suggestion?

**Comment by beck-8** [comment]
It has not been resolved.
```
root@localhost:~# minio -v
minio version RELEASE.2023-12-13T23-28-55Z (commit-id=3781a0f9ad47ed77d1050e1b36848e2ebcf73194)
Runtime: go1.21.5 linux/amd64
License: GNU AGPLv3 <https://www.gnu.org/licenses/agpl-3.0.html>
Copyright: 2015-2023 MinIO, Inc.
```
```
UTC 12/18/2023
ecea552-4e06-4a76-9180-e584e018a05a
resources are insufficient for the write operation .minio.sys/buckets/xxx/.usage-cache.bin (cmd.InsufficientWriteQuorum)
nal/logger/logger.go:259:logger.LogIf()
rasure.go:561:cmd.erasureObjects.nsScanner.func2()
```

**Comment by mehransaeed7810** [comment]
Same issue here. minio pod complains about insufficient number of drives available. however got two drives with 180Gi each and more than enough space. still no luck with the resolution

**Reply by valkiriaaquatica** [reply]
> Same issue here. minio pod complains about insufficient number of drives available. however got two drives with 180Gi each and more than enough space. still no luck with the resolution

Hi, what type of StorageClass are you using?
At least in my case I was using  NFS with a NAS and it was a problem of permissions, in the logs of the pod just "seem" to appeared the error attach above but the first ones was a permission error. Can you check when you initializate the pod what logs is it showing?
Thanks!

**Reply by TheOnlyWayUp** [reply]
This was helpful, my sshfs mount disappeared and I didn't realize it till reading your message.

**Comment by chakraborty29** [comment]
I had this issue on a k8s install, where I lowered the default resources limits and requests. Setting it back to the default in the helm values resolved the issue for me.
